crates/shared/src/persistence/sqlite.rs

```diff
@@ -0,0 +1,211 @@
+use super::BuilderPersistence;
+use anyhow::Result;
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use sqlx::Row;
+use sqlx::SqlitePool;
+use std::time::{Instant, SystemTime};
+
+#[allow(unused_imports)]
+use super::get_sqlite_test_db_path;
+
+#[derive(Debug)]
+pub struct SqliteTxnDb {
+    pool: SqlitePool,
+}
```

Comment:
In the future we're going to add some pruning parameters here like minimum retention or target retention. It will be expanded later but I don't have a strong opinion for now.